frontend/lib/Repository/AdminRepository/admin_groups_repository.dart

```diff
@@ -6,255 +6,191 @@ import '../../DataModels/backend_response.dart';
 import '../../DataModels/group_endpoints.dart';
 import '../../DataModels/group_summary.dart';
 import '../../DataModels/user_summary.dart';
-import '../UserRepository/auth_gateway.dart';
 import '../UserRepository/user_gateway.dart';
+import '../rest_client.dart';
 
 class AdminGroupsRepository {
-  Dio _client = Dio();
+  final RestClient _client = RestClient();
   UserGateway userGateway = UserGateway();
```

Comment:
```suggestion
  final RestClient _client = RestClient();
```

Przeleciałbym jeszcze raz po klasach, gdzie dodaliśmy `RestClient` i pousuwałbym `UserGateway`, jeżeli nie jest już więcej potrzebny